### DISCUSSION TITLE
Make sharing feature useful

### DISCUSSION THREAD

**Discussion by cmot17** [octokatherine/word-master#34] [Ideas]
Demo: https://cmot17.github.io/word-master
Repo: https://github.com/cmot17/word-master/tree/share-link

This isn't 100% complete yet but it should work. The sharing feature shares a link that allows other people to play the same puzzle without spoilers.

https://cmot17.github.io/word-master#/ZCJHE 4/6

⬛⬛⬛🟩🟩
⬛⬛🟨⬛⬛
⬛🟨⬛⬛🟨
🟩🟩🟩🟩🟩

I did have to add a decent amount of code because of the need for react-router to encode data in the URL.

**Comment by octokatherine** [comment]
this is a pretty cool idea, let me explore it a little more

**Reply by octokatherine** [reply]
I do like this feature if we have a link to play the same word

**Reply by octokatherine** [reply]
I would only add the share button when they win (right now it shows on the lost modal)

**Reply by cmot17** [reply]
It might be helpful to share a loss, for instance, to share the result to somebody who sent you the puzzle link, or to share a link to a hard puzzle to see if others can solve it. Wordle allows the same, IIRC.

**Reply by octokatherine** [reply]
okay yeah that works

**Reply by cmot17** [reply]
Should I open a PR with these changes?

**Reply by octokatherine** [reply]
sure!

**Comment by Amadeus-** [comment]
I was just curious if the PR for this update to the sharing concept was rejected, or am I just not seeing it? :)

**Reply by cmot17** [reply]
Sorry, I've been super busy recently - I'll open the PR as soon as I have the free time.
